File: scripts/remove-console-logs.js
#!/usr/bin/env node

/**
 * 批量移除 console.log 调试日志
 * 保留 console.error 和 console.warn（用于错误追踪和警告）
 */

const fs = require('fs');
const path = require('path');
const { execSync } = require('child_process');

// 需要处理的目录
const directories = ['app', 'components', 'lib'];

// 统计信息
let totalRemoved = 0;
let filesProcessed = 0;
let filesModified = 0;

/**
 * 递归查找所有 .ts 和 .tsx 文件
 */
function findFiles(dir, fileList = []) {
  const files = fs.readdirSync(dir);
  
  files.forEach(file => {
    const filePath = path.join(dir, file);
    const stat = fs.statSync(filePath);
    
    if (stat.isDirectory()) {
      // 跳过 node_modules 和 .next
      if (file !== 'node_modules' && file !== '.next') {
        findFiles(filePath, fileList);
      }
    } else if (file.endsWith('.ts') || file.endsWith('.tsx')) {
      fileList.push(filePath);
    }
  });
  
  return fileList;
}

/**
 * 移除文件中的 console.log
 */
function removeConsoleLogs(filePath) {
  let content = fs.readFileSync(filePath, 'utf8');
  const originalContent = content;
  let removedCount = 0;
  
  // 匹配 console.log 的各种情况
  // 1. 单独一行的 console.log(...);
  // 2. 在代码中间的 console.log(...);
  // 3. console.log 前后可能有注释
  
  // 使用正则表达式匹配并移除 console.log 语句
  // 匹配模式：console.log(...); 包括多行的情况
  const consoleLogPattern = /^\s*console\.log\([^;]*\);\s*$/gm;
  const multiLinePattern = /console\.log\([^;)]*\);?/g;
  
  // 先处理单独一行的 console.log
  content = content.replace(consoleLogPattern, '');
  
  // 处理在代码中间的 console.log（但保留在同一行的情况）
  // 需要小心处理，避免破坏代码结构
  const lines = content.split('\n');
  const newLines = [];
  
  for (let i = 0; i < lines.length; i++) {
    const line = lines[i];
    
    // 检查是否包含 console.log（但不在注释中）
    if (line.includes('console.log') && !line.trim().startsWith('//')) {
      // 检查是否是完整的 console.log 语句
      if (line.trim().endsWith(';') && line.includes('console.log(')) {
        // 尝试移除 console.log，但要小心处理
        // 如果整行都是 console.log，则移除整行
        const trimmed = line.trim();
        if (trimmed.startsWith('console.log(') && trimmed.endsWith(';')) {
          removedCount++;
          continue; // 跳过这一行
        }
        
        // 如果 console.log 在行的中间或末尾
        // 使用更保守的方法：只移除完整的 console.log(...); 语句
        const beforeLog = line.substring(0, line.indexOf('console.log'));
        const afterLog = line.substring(line.indexOf('console.log'));
        
        // 尝试找到 console.log(...); 的结束位置
        let logStart = afterLog.indexOf('console.log(');
        if (logStart !== -1) {
          let depth = 0;
          let logEnd = logStart + 'console.log('.length;
          let inString = false;
          let stringChar = '';
          
          for (let j = logEnd; j < afterLog.length; j++) {
            const char = afterLog[j];
            
            if (!inString && (char === '"' || char === "'" || char === '`')) {
              inString = true;
              stringChar = char;
            } else if (inString && char === stringChar && afterLog[j - 1] !== '\\') {
              inString = false;
            } else if (!inString) {
              if (char === '(') depth++;
              if (char === ')') {
                if (depth === 0) {
                  logEnd = j + 1;
                  break;
                }
                depth--;
              }
            }
          }
          
          // 检查后面是否有分号
          if (afterLog[logEnd] === ';') {
            const newLine = beforeLog + afterLog.substring(logEnd + 1);
            removedCount++;
            newLines.push(newLine.trim() ? newLine : '');
            continue;
          }
        }
      }
    }
    
    newLines.push(line);
  }
  
  content = newLines.join('\n');
  
  // 清理多余的空行（连续 3 个或更多空行变为 2 个）
  content = content.replace(/\n{3,}/g, '\n\n');
  
  if (content !== originalContent) {
    fs.writeFileSync(filePath, content, 'utf8');
    filesModified++;
    totalRemoved += removedCount || 1;
  }
  
  filesProcessed++;
}

// 主函数
function main() {
  console.log('🚀 开始批量移除 console.log...\n');
  
  directories.forEach(dir => {
    if (!fs.existsSync(dir)) {
      console.log(`⚠️  目录不存在: ${dir}`);
      return;
    }
    
    console.log(`📁 处理目录: ${dir}`);
    const files = findFiles(dir);
    console.log(`   找到 ${files.length} 个文件\n`);
    
    files.forEach(file => {
      removeConsoleLogs(file);
    });
  });
  
  console.log('\n✅ 处理完成！');
  console.log(`   处理文件数: ${filesProcessed}`);
  console.log(`   修改文件数: ${filesModified}`);
  console.log(`   移除的 console.log 数量: ${totalRemoved}`);
}

main();

File: scripts/remove-console-logs.py
#!/usr/bin/env python3
"""
批量移除 console.log 调试日志
保留 console.error 和 console.warn
"""

import os
import re
import sys
from pathlib import Path

# 需要处理的目录
DIRECTORIES = ['app', 'components', 'lib']

# 统计信息
total_removed = 0
files_processed = 0
files_modified = 0

def find_files(directories):
    """递归查找所有 .ts 和 .tsx 文件"""
    files = []
    for directory in directories:
        if not os.path.exists(directory):
            print(f"⚠️  目录不存在: {directory}")
            continue
        
        for root, dirs, filenames in os.walk(directory):
            # 跳过 node_modules 和 .next
            dirs[:] = [d for d in dirs if d not in ['node_modules', '.next']]
            
            for filename in filenames:
                if filename.endswith(('.ts', '.tsx')):
                    files.append(os.path.join(root, filename))
    
    return files

def remove_console_logs(file_path):
    """移除文件中的 console.log"""
    global total_removed, files_processed, files_modified
    
    try:
        with open(file_path, 'r', encoding='utf-8') as f:
            content = f.read()
        
        original_content = content
        removed_count = 0
        
        # 按行处理
        lines = content.split('\n')
        new_lines = []
        i = 0
        
        while i < len(lines):
            line = lines[i]
            original_line = line
            
            # 跳过注释行（但保留）
            if line.strip().startswith('//'):
                new_lines.append(line)
                i += 1
                continue
            
            # 检查是否包含 console.log
            if 'console.log(' in line:
                # 检查是否是独立的 console.log 语句
                # 匹配：前面只有空白，后面是 console.log(...);
                stripped = line.strip()
                
                # 情况1：整行就是 console.log(...);
                if re.match(r'^\s*console\.log\([^;]*\);\s*$', line):
                    removed_count += 1
                    i += 1
                    continue  # 跳过这一行
                
                # 情况2：console.log(...); 在行尾，前面有其他代码（通常是调试用的）
                # 例如：someCode; console.log(...);
                # 这种情况我们也要移除 console.log 部分
                if re.search(r'console\.log\([^;)]*\);\s*$', line):
                    # 移除 console.log 部分
                    line = re.sub(r'[;\s]*console\.log\([^;)]*\);\s*$', '', line)
                    removed_count += 1
                    if line.strip():  # 如果还有内容，保留
                        new_lines.append(line.rstrip())
                    i += 1
                    continue
                
                # 情况3：console.log 在多行中（复杂情况，暂时保留原样）
                # 这种情况比较复杂，需要处理括号匹配
                # 为了安全，暂时跳过
                new_lines.append(line)
                i += 1
                continue
            
            new_lines.append(line)
            i += 1
        
        # 重新组合内容
        new_content = '\n'.join(new_lines)
        
        # 清理多余的空行（连续3个或更多空行变为2个）
        new_content = re.sub(r'\n{3,}', '\n\n', new_content)
        
        # 如果内容有变化，写回文件
        if new_content != original_content:
            with open(file_path, 'w', encoding='utf-8') as f:
                f.write(new_content)
            files_modified += 1
            total_removed += removed_count
        
        files_processed += 1
        
    except Exception as e:
        print(f"❌ 处理 {file_path} 时出错: {e}", file=sys.stderr)

def main():
    """主函数"""
    print('🚀 开始批量移除 console.log...\n')
    
    files = find_files(DIRECTORIES)
    print(f'📁 找到 {len(files)} 个文件\n')
    
    for file_path in files:
        remove_console_logs(file_path)
    
    print('\n✅ 处理完成！')
    print(f'   处理文件数: {files_processed}')
    print(f'   修改文件数: {files_modified}')
    print(f'   移除的 console.log 数量: {total_removed}')

if __name__ == '__main__':
    main()

